exporting/mongodb/mongodb.c

```diff
@@ -0,0 +1,310 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#define EXPORTING_INTERNALS
+#include "mongodb.h"
+
+#define CONFIG_FILE_LINE_MAX ((CONFIG_MAX_NAME + CONFIG_MAX_VALUE + 1024) * 2)
+
+/**
+ * Initialize MongoDB connector specific data, including a ring buffer
+ *
+ * @param instance an instance data structure.
+ * @return Returns 0 on success, 1 on failure.
+ */
+int mongodb_init(struct instance *instance)
+{
+    struct mongodb_specific_config *connector_specific_config = instance->config.connector_specific_config;
+    mongoc_uri_t *uri;
+    bson_error_t error;
+
+    if (unlikely(!connector_specific_config->collection || !*connector_specific_config->collection)) {
+        error("EXPORTING: collection name is a mandatory MongoDB parameter, but it is not configured");
+        return 1;
+    }
+
+    uri = mongoc_uri_new_with_error(instance->config.destination, &error);
+    if (unlikely(!uri)) {
+        error("EXPORTING: failed to parse URI: %s. Error message: %s", instance->config.destination, error.message);
+        return 1;
+    }
+
+    int32_t socket_timeout =
+        mongoc_uri_get_option_as_int32(uri, MONGOC_URI_SOCKETTIMEOUTMS, instance->config.timeoutms);
+    if (!mongoc_uri_set_option_as_int32(uri, MONGOC_URI_SOCKETTIMEOUTMS, socket_timeout)) {
```

Comment:
You called the same function two times, I believe you wanted to test `socket_timeout` here.

Reply:
They are different functions - __get__ and __set__.